### PR TITLE
[cmake] add fcgi option to enable FastCGI compilation, default OFF

### PR DESCRIPTION
**Pull request by linev**
This is very special component of THttpServer, which is not used up to
now. Make it as cmake option and disable by default. Only when really
needed, user can enable it.

### CODE REVIEW

**Comment by linev** [comment]
Was requested by @osschar to make this option by default OFF. 
There is problem to include `libfcgi.so` into CMS packaging

**Comment by linev** [comment]
@amadio 
There are many no-longer-used flags in config/Makefile.in file.
Maybe one could just remove them?

File: cmake/modules/FindFastCGI.cmake [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Copyright (C) 1995-2019, Rene Brun and Fons Rademakers.
+# All rights reserved.
+#
+# For the licensing terms see $ROOTSYS/LICENSE.
+# For the list of contributors see $ROOTSYS/README/CREDITS.
+
+# Find the FastCGI includes and library
+#
+#  FASTCGI_INCLUDE_DIR - where to find fcgiapp.h
+#  FASTCGI_LIBRARY     - librart when using MySQL.
```
Comment (amadio) [inline cmake/modules/FindFastCGI.cmake]: Typo, and this is not MySQL...

**Comment by linev** [comment]
@amadio I provide requested changes. 
Question - why one needs `FASTCGI_INCLUDE_DIRS` and `FASTCGI_LIBRARIES` as uncached variables? Should I use them when building THttpServer?

File: cmake/modules/FindFastCGI.cmake [hunk 2]
```diff
@@ -7,41 +7,28 @@
...
-#  FASTCGI_LIBRARY     - librart when using MySQL.
-#  FASTCGI_FOUND       - True if FASTCGI found.
+#  FASTCGI_LIBRARY     - library when using FastCGI.
+#  FASTCGI_FOUND       - true if FASTCGI found.
 
-if(FASTCGI_INCLUDE_DIR OR FASTCGI_)
-  # Already in cache, be silent
-  SET(FASTCGI_FIND_QUIETLY TRUE)
+if(NOT FASTCGI_INCLUDE_DIR)
+   find_path(FASTCGI_INCLUDE_DIR NAME fcgiapp.h PATH_SUFFIXES include)
```
Comment (amadio) [inline cmake/modules/FindFastCGI.cmake]: This `if` is not necessary, as if the variable is already in the cache, the command will not do anything.

**Comment by amadio** [comment]
@linev Thanks for the changes. Yes, usually one uses the uncached variables in the build, and CMake modules provide them because maybe you will need more than one include dir, or to add an include dir for a dependency, etc. Not that it was broken before, but this way is more robust to changes in fcgi=ON/OFF. As for config/Makefile.in, we should ask Axel. In ROOT it's not used anymore after the configure-based build system was removed. We could just remove the file, but I think that dependent projects still use it.

**Comment by amadio** [comment]
I think that the fixup commits could be squashed on merge. There is no point in commiting a typo to then fix in a later commit, for example. Better to commit the final changes directly.

**Comment by linev** [comment]
> In ROOT it's not used anymore after the configure-based build system was removed. 
> We could just remove the file, but I think that dependent projects still use it.

Most probably, file used by external projects. But one can remove many ROOT-internal variables - like fastcgi include paths. Probably one can keep just list of build components.

> I think that the fixup commits could be squashed on merge

Yes, you can do it.

**Comment by linev** [comment]
Failing tests has nothing to do with fcgi compilation

**Pull request merged**